### PR TITLE
Substrate add new fields in lab certification and services pallet.

### PR DESCRIPTION
**Pull request by agustinustheo**
### JIRA Link
https://blocksphere2020.atlassian.net/browse/DBIO-342?atlOrigin=eyJpIjoiMWE5MzQ1ZGMzMjM3NGE1ZThmYmEzN2Q1N2IyMGY2YTAiLCJwIjoiaiJ9

https://blocksphere2020.atlassian.net/browse/DBIO-341?atlOrigin=eyJpIjoiMmM0NDhlNGZiZjA2NGJiYTlmMzBhNmVhNzA5NTRhYTciLCJwIjoiaiJ9

### Changelog / Description
Substrate | Update info fields in Service struct
Substrate | Certification lab add field for supporting document

- Add enum `CurrencyType`
- Add enum `DurationType`
- Add `supporting_document` field in lab certifications
- Add `total_price` field in `PriceByCurrency` struct
- Add `test_result_sample` field in services
- Add `dna_collection_process` field in services
- Changed `prices` field to `price_components`
- Changed `key` field in `Prices` to `component`
- Update `types.json`

#### BREAKING CHANGE: 
- New `CurrencyType` and `DurationType` enum added to `types.json`, other projects (FE, BE, Faucet) that communicate with `debio-node` need to update their `types.json`.
- Changed field names so other projects need to update accordingly.

### CODE REVIEW

**Comment by agustinustheo** [comment]
Please rebase if done @kevinjanada @SHA888

File: pallets/services/src/lib.rs [hunk 1]
```diff
@@ -222,8 +225,21 @@ impl<T: Config> ServiceInterface<T> for Pallet<T> {
 
         let owner_service_count = <Self as ServiceInterface<T>>::services_count_by_owner(owner_id);
         let service_id = Self::generate_service_id(owner_id, owner_service_count);
+
+        // Calculate total price
+        let mut service_info_mut = service_info.clone();
+        for (idx, price_by_currency) in service_info.prices_by_currency.iter().enumerate() {
+            service_info_mut.prices_by_currency[idx].total_price -= price_by_currency.total_price;
```
Comment (kevinjanada) [inline pallets/services/src/lib.rs:232]: Bagian ini memang hrs di kurang dulu yah? @agustinustheo

Comment (agustinustheo) [reply pallets/services/src/lib.rs:232]: Yea, antara itu atau inisialisasi ulang menjadi 0. Mending yang mana? @kevinjanada

**Review by kevinjanada — APPROVED** [review]

**Comment by SHA888** [comment]
LGTM

**Pull request merged**